qiskit/transpiler/passes/optimization/remove_diagonal_gates_before_measure.py

```diff
@@ -48,22 +37,4 @@ def run(self, dag):
         Returns:
             DAGCircuit: the optimized DAG.
         """
-        diagonal_1q_gates = (RZGate, ZGate, TGate, SGate, TdgGate, SdgGate, U1Gate)
-        diagonal_2q_gates = (CZGate, CRZGate, CU1Gate, RZZGate)
-
-        nodes_to_remove = set()
-        for measure in dag.op_nodes(Measure):
-            predecessor = next(dag.quantum_predecessors(measure))
-
-            if isinstance(predecessor, DAGOpNode) and isinstance(predecessor.op, diagonal_1q_gates):
-                nodes_to_remove.add(predecessor)
-
-            if isinstance(predecessor, DAGOpNode) and isinstance(predecessor.op, diagonal_2q_gates):
-                successors = dag.quantum_successors(predecessor)
-                if all(isinstance(s, DAGOpNode) and isinstance(s.op, Measure) for s in successors):
-                    nodes_to_remove.add(predecessor)
-
-        for node_to_remove in nodes_to_remove:
-            dag.remove_op_node(node_to_remove)
-
-        return dag
+        remove_diagonal_gates_before_measure(dag)
```

Comment:
You need to return the dag here, the pass interface expects a dag object to returned, even if we mutate by reference.

```suggestion
        remove_diagonal_gates_before_measure(dag)
        return dag
```